subiquity/server/ubuntu_drivers.py

```diff
@@ -168,6 +172,129 @@ async def list_oem(self, root_dir: str, context) -> List[str]:
         return self._oem_metapackages_from_output(result.stdout.decode("utf-8"))
 
 
+class UbuntuDriversFakePCIDevicesInterface(UbuntuDriversInterface):
+    """An implementation of ubuntu-drivers that wraps the calls with
+    the umockdev wrapper script.
+
+
+    Requires an online install to download umockdev packages or
+    a modified ISOs with the packages added to the pool.
+    """
+
+    def __init__(self, app, gpgpu: bool) -> None:
+        super().__init__(app, gpgpu)
+
+        # Default config. A broadcom wireless device.
+        self.dev_config = {
+            "devices": [
+                {
+                    "modalias": "pci:v000014E4d00004353sv00sd01bc02sc80i00",
+                    "vendor": "0x14E4",
+                    "device": "0x4353",
+                }
+            ]
+        }
+
+        self.dev_config_path = "/tmp/umockdev_config.yaml"
+
+        # write config to live environment
+        write_file(self.dev_config_path, yaml.safe_dump(self.dev_config), mode=0o777)
```

Comment:
The vendor and device values can be read from the modalias - note the `v` and `d` values at the beginning of the modalias string matching the vendor and device - which means that we could do a kernel command line custom string of `subiquity-fake-pci-devices=modalias` and get driver fakery in return without slimy snap cycle.

Reply:
Ah I hadn't realized, nice. I do like that idea more than changing with early-commands.